src/Native/Runtime/MathHelpers.cpp

```diff
@@ -82,6 +82,70 @@ EXTERN_C REDHAWK_API UInt64 REDHAWK_CALLCONV RhpDbl2ULng(double val)
     return((UInt64)val);
 }
 
+// CORERT Specific - on Project N the arguments to these helpers are inverted
+#ifdef CORERT
+#include <cmath>
+
+#define CLR_NAN_32 0xFFC00000
+#define CLR_NAN_64 0xFFF8000000000000L
+
+EXTERN_C REDHAWK_API float REDHAWK_CALLCONV RhpFltRem(float dividend, float divisor)
+{
+    //
+    // From the ECMA standard:
+    //
+    // If [divisor] is zero or [dividend] is infinity
+    //   the result is NaN.
+    // If [divisor] is infinity,
+    //   the result is [dividend] (negated for -infinity***).
+    //
+    // ***"negated for -infinity" has been removed from the spec
+    //
+
+    if (divisor==0 || !isfinite(dividend))
+    {
+        UInt32 NaN = CLR_NAN_32;
+        return *(float *)(&NaN);
```

Comment:
clang does not seem to complain about those casts, but gcc will raise:

a.c:6:5: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]

It is usually better to use a local union with both a UInt32 and float and assign the int and return the float.